File: LeetCodeProblemsCpp/LeetCodeProblemsCpp/Problems/LeetCode1863SumOfAllSubsetXORTotals.cpp
#include "LeetCode1863SumOfAllSubsetXORTotals.h"
using  namespace std;
void Backtrack(int index,vector<int>& nums,int currentXor,int& sum,int length)
{
    int temp{};
    for (int i=index; i < length; i++)
    {
        temp = currentXor^ nums[i];
        sum += temp;
        Backtrack(i + 1, nums, temp, sum, length);
    }
}
int LeetCode1863SumOfAllSubsetXORTotals::subsetXORSum(vector<int>& nums)
{
    int sum{};
    Backtrack(0, nums, 0, sum, nums.size());
    return sum;
}

File: LeetCodeProblemsGo/LeetCode1863SumOfAllSubsetXORTotals/LeetCode1863SumOfAllSubsetXORTotals.go
package LeetCode1863SumOfAllSubsetXORTotals

func subsetXORSum(nums []int) int {
	sum := 0
	length := len(nums)
	backtrack(0, nums, 0, &sum, length)
	return sum
}
func backtrack(index int, nums []int, currentXor int, sum *int, length int) {
	temp := 0
	for i := index; i < length; i++ {
		temp = currentXor ^ nums[i]
		*sum += temp
		backtrack(i+1, nums, temp, sum, length)
	}
}
